DigitalLearningSolutions.Web.Tests/Controllers/Register/RegisterAdminControllerTests.cs

```diff
@@ -0,0 +1,214 @@
+            // Given
+            const int centreId = 7;
+            A.CallTo(() => centresDataService.GetCentreName(centreId)).Returns("Some centre");
+            A.CallTo(() => centresDataService.GetCentreAutoRegisterValues(centreId)).Returns((false, "email@email"));
+
+            // When
+            var result = controller.Index(centreId);
+
+            // Then
+            A.CallTo(() => centresDataService.GetCentreName(centreId)).MustHaveHappened(1, Times.Exactly);
+            A.CallTo(() => centresDataService.GetCentreAutoRegisterValues(centreId)).MustHaveHappened(1, Times.Exactly);
+            var data = controller.TempData.Peek<RegistrationData>()!;
+            data.Centre.Should().Be(centreId);
+            result.Should().BeRedirectToActionResult().WithActionName("PersonalInformation");
+        }
+
+        [Test]
+        public void PersonalInformationPost_with_wrong_autoregisteremail_for_centre_fails_validation()
+        {
+            // Given
+            const int centreId = 7;
+            var model = new PersonalInformationViewModel
+            {
+                FirstName = "Test",
+                LastName = "User",
+                Centre = centreId,
+                Email = "wrong@email"
+            };
+            var data = new RegistrationData { Centre = centreId };
+            controller.TempData.Set(data);
+            A.CallTo(() => centresDataService.GetCentreAutoRegisterValues(centreId)).Returns((false, "right@email"));
+
+            // When
+            var result = controller.PersonalInformation(model);
+
+            // Then
+            A.CallTo(() => centresDataService.GetCentreAutoRegisterValues(centreId)).MustHaveHappened(1, Times.Exactly);
+            result.Should().BeViewResult().WithDefaultViewName();
+        }
+
+        [Test]
+        public void PersonalInformationPost_with_email_already_in_use_fails_validation()
+        {
+            // Given
+            const int centreId = 7;
+            const string email = "right@email";
+            var model = new PersonalInformationViewModel
+            {
+                FirstName = "Test",
+                LastName = "User",
+                Centre = centreId,
+                Email = email
+            };
+            var data = new RegistrationData { Centre = centreId };
+            controller.TempData.Set(data);
+            A.CallTo(() => centresDataService.GetCentreAutoRegisterValues(centreId)).Returns((false, email));
+            A.CallTo(() => userDataService.GetAdminUserByEmailAddress(email)).Returns(new AdminUser());
+
+            // When
+            var result = controller.PersonalInformation(model);
+
+            // Then
+            A.CallTo(() => centresDataService.GetCentreAutoRegisterValues(centreId)).MustHaveHappened(1, Times.Exactly);
+            A.CallTo(() => userDataService.GetAdminUserByEmailAddress(email)).MustHaveHappened(1, Times.Exactly);
+            result.Should().BeViewResult().WithDefaultViewName();

```

Comment:
Again it would be great if we could check that the Email field specifically failed validation